usecases/config/environment.go

```diff
@@ -479,9 +480,9 @@ func FromEnv(config *Config) error {
 		config.EnableApiBasedModules = true
 	}
 
-	config.AutoSchema.Enabled = true
+	config.AutoSchema.Enabled = runtime.NewDynamicValue(true)
 	if v := os.Getenv("AUTOSCHEMA_ENABLED"); v != "" {
-		config.AutoSchema.Enabled = !(strings.ToLower(v) == "false")
+		config.AutoSchema.Enabled.SetDefault(!(strings.ToLower(v) == "false"))
 	}
```

Comment:
If you would do a pattern like this, you could ditch the `SetDefault` method and all the locks around it:
```go
autoschemaEnabledDefault := true
if v := os.Getenv("AUTOSCHEMA_ENABLED"); v != "" {
    autoschemaEnabledDefault = !(strings.ToLower(v) == "false")
}
config.AutoSchema.Enabled = runtime.NewDynamicValue(autoschemaEnabledDefault)
```

Reply:
Done. also removed the need for `SetDefault` (along with it's lock)